### PR TITLE
Fix tls connect event is "secureConnect"

### PR DESCRIPTION
**Pull request by jeffjen**
[See API page](https://nodejs.org/api/tls.html#tls_class_tls_tlssocket_1)

TLS stream upon connection emits "secureConnect" event.

Tested against *stunnel*

### CODE REVIEW

**Comment by luin** [comment]
Doesn't tls stream also emit `connect` event?

**Comment by jeffjen** [comment]
Well it looks like v4.1.2 had the tls connection emit "connect" as well as "secureConnect"

For the node build from https://deb.nodesource.com/setup, which had version `v0.10.40`, it doesn't emit "connect", only emits "secureConnect".

Consider this fix for backwards compatibility then.

File: lib/redis.js [hunk 1]
```diff
@@ -256,13 +256,17 @@ Redis.prototype.connect = function (callback) {
         reject(err);
         return;
       }
+      var CONNECT_EVENT = 'connect';
+      if (_this.options.tls) {
+          CONNECT_EVENT = 'secureConnect';
+      }
```
Comment (luin) [inline lib/redis.js]: It would be better to use `var CONNECT_EVENT = _this.options.tls ? 'secureConnect' : 'connect';` instead

**Comment by luin** [comment]
Good to know:-). According to the [issue](https://github.com/nodejs/node-v0.x-archive/issues/6903), it seems that `secureConnect` is more preferable than `connect`. I just wrote a comment

**Comment by jeffjen** [comment]
Pushed a change to do tertiary assignment to CONNECT_EVENT.

I know its not recommended to connect via TLS to REDIS since the support of it is minimal (need SSL proxy to do it), but hopefully this will payout in the long run.

**Comment by luin** [comment]
Thank you for the patch! Merged :rocket:

**Comment by luin** [comment]
Shipped in 1.10.0

**Pull request merged**